test/python/test_rticonnextdds_discovery.py

```diff
@@ -0,0 +1,195 @@
+
+    # Calling again to wait_for_match should timeout
+    with pytest.raises(rti.TimeoutError) as excinfo:
+      the_output.wait_for_match(1000)
+
+    matches = the_output.get_matched_inputs()
+    assert {'name': 'MyReader'} in matches
+    assert {'name': 'TestReader'} in matches
+
+  def test_multiple_outputs(self, discovery_connector, discovery_writer_only_output):
+    the_input = discovery_connector.get_input("MySubscriber::MyReader")
+
+    total_matches = 0
+    # the_input should match two outpouts, the one from discovery_writer_only_output
+    # and the one defined within discovery_connector
+    while total_matches < 2:
+      total_matches += the_input.wait_for_match(3000)
+    assert total_matches == 2
+
+    # Calling again to wait_for_match should timeout
+    with pytest.raises(rti.TimeoutError) as excinfo:
+      the_input.wait_for_match(1000)
+
+    matches = the_input.get_matched_outputs()
+    assert {'name': 'MyWriter'} in matches
+    assert {'name': 'TestWriter'} in matches
+
+  # Even though we are going to use the discovery_reader_only_input, don't use the
+  # pytest fixture as we want to control the deletion of it from within the
+  # test function
+  def test_unmatched_input(self, discovery_writer_only_output):
+    # To begin with, no matching occurs
+    assert None is discovery_writer_only_output.get_matched_inputs()
+
+    # Create the Connector object containing the matching input
+    reader_connector = rti.Connector(
+        "MyParticipantLibrary::DiscoveryTestReaderOnly",
+        os.path.join(os.path.dirname(os.path.realpath(__file__)),"../xml/TestConnector.xml"))
+    the_input = reader_connector.get_input("TestSubscriber::TestReader")
+    assert the_input is not None
+
+    # Check that matching occurs
+    change_in_matches = discovery_writer_only_output.wait_for_match()
+    assert change_in_matches == 1
+    matches = discovery_writer_only_output.get_matched_inputs()
+    assert {'name': 'TestReader'} in matches
+    change_in_matches = the_input.wait_for_match()
+    assert change_in_matches == 1
+    matches = the_input.get_matched_outputs()
+    assert {'name': 'TestWriter'} in matches
+
+    # If we now delete the reader_connector object which we created above, they
+    # will unmatch
+    reader_connector.close()
+    change_in_matches = discovery_writer_only_output.wait_for_match()
+    assert change_in_matches == -1
+    matches = discovery_writer_only_output.get_matched_inputs()
+    assert matches is None
+
+  # Even though we are going to use the discovery_writer_only_output, don't use the
+  # pytest fixture as we want to control the deletion of it from within the
+  # test function
+  def test_unmatched_output(self, discovery_reader_only_input):
+    # To begin with, no matching occurs
+    assert None is discovery_reader_only_input.get_matched_outputs()
+
+    # Create the Connector object containing the matching input
+    writer_connector = rti.Connector(
```

Comment:
Any don't you use the `discovery_writer_only_connector` fixture?

Reply:
I wanted to be able to force entities to unmatch so I could test the APIs in that case. If i use the fixture I couldn't find a neat way of deleting the connector.